src/test/java/seedu/address/logic/commands/FindOrderCommandTest.java

```diff
@@ -0,0 +1,90 @@
+import static seedu.address.testutil.TypicalOrders.SALESORDER2;
+import static seedu.address.testutil.TypicalOrders.SALESORDER3;
+import static seedu.address.testutil.TypicalOrders.getTypicalOrderBook;
+import static seedu.address.testutil.TypicalPersons.getTypicalAddressBook;
+import static seedu.address.testutil.TypicalTasks.getTypicalTaskBook;
+
+import java.util.Arrays;
+import java.util.Collections;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.order.OrderContainsKeywordsPredicate;
+
+/**
+ * Contains integration tests (interaction with the Model) for {@code FindOrderCommand}.
+ * Modelled after FindCommandTest
+ */
+class FindOrderCommandTest {
+
+    private Model model = new ModelManager(getTypicalAddressBook(), getTypicalTaskBook(),
+            getTypicalOrderBook(), new UserPrefs());
+    private Model expectedModel = new ModelManager(getTypicalAddressBook(),
+            getTypicalTaskBook(), getTypicalOrderBook(), new UserPrefs());
+
+
+    @Test
+    public void equals() {
+        OrderContainsKeywordsPredicate firstPredicate =
+                new OrderContainsKeywordsPredicate(Collections.singletonList("first"));
+        OrderContainsKeywordsPredicate secondPredicate =
+                new OrderContainsKeywordsPredicate(Collections.singletonList("second"));
+
+        FindOrderCommand findFirstCommand = new FindOrderCommand(firstPredicate);
+        FindOrderCommand findSecondCommand = new FindOrderCommand(secondPredicate);
+
+        // same object -> returns true
+        assertTrue(findFirstCommand.equals(findFirstCommand));
+
+        // same values -> returns true
+        FindOrderCommand findFirstCommandCopy = new FindOrderCommand(firstPredicate);
+        assertTrue(findFirstCommand.equals(findFirstCommandCopy));
+
+        // different types -> returns false
+        assertFalse(findFirstCommand.equals(1));
+
+        // null -> returns false
+        assertFalse(findFirstCommand.equals(null));
+
+        // different order -> returns false
+        assertFalse(findFirstCommand.equals(findSecondCommand));
+    }
+
+    @Test
+    public void execute_zeroKeywords_noOrderFound() {
+        String expectedMessage = String.format(MESSAGE_ORDERS_LISTED_OVERVIEW, 0);
+        OrderContainsKeywordsPredicate predicate = preparePredicate(" ");
+        FindOrderCommand command = new FindOrderCommand(predicate);
+        expectedModel.updateFilteredOrderList(predicate);
+        assertCommandSuccess(command, model, expectedMessage, expectedModel);
+        assertEquals(Collections.emptyList(), model.getFilteredOrderList());
+    }
+
+    @Test
+    public void execute_multipleKeywords_multipleOrdersFound() {
+        String expectedMessage = String.format(MESSAGE_ORDERS_LISTED_OVERVIEW, 3);
+        OrderContainsKeywordsPredicate predicate = preparePredicate("Josh Mac Clark");
```

Comment:
Should finding orders by other fields (label, salesId, date - not sure if this is supported) also be tested?

Reply:
edited the test case to demonstrate